Retry webhooks to tarantool.io in the docs deployment

tarantool.io can sometimes return non-200 code to webhooks due to it being redeployed or some other internal reasons. A simple retry in a few seconds should be successful. However, our pipelines can't handle this situation. 

I propose using https://github.com/marketplace/actions/retry-step.

---

https://github.com/tarantool/doc/actions/runs/3484844257/jobs/5829815120

```console
$ curl --fail --show-error -v --data '{"update_key":"***"}' --header 'Content-Type: application/json' --request POST ***latest/
Deploy version latest on the server
Note: Unnecessary use of -X or --request, POST is already inferred.
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed

  0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0
  0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0
  0     0    0     0    0     0      0      0 --:--:--  0:00:01 --:--:--     0
  0     0    0     0    0     0      0      0 --:--:--  0:00:02 --:--:--     0
  0     0    0     0    0     0      0      0 --:--:--  0:00:03 --:--:--     0
  0     0    0     0    0     0      0      0 --:--:--  0:00:04 --:--:--     0
  0     0    0     0    0     0      0      0 --:--:--  0:00:05 --:--:--     0
  0     0    0     0    0     0      0      0 --:--:--  0:00:06 --:--:--     0
  0     0    0     0    0     0      0      0 --:--:--  0:00:07 --:--:--     0
  0     0    0     0    0     0      0      0 --:--:--  0:00:08 --:--:--     0
  0     0    0     0    0     0      0      0 --:--:--  0:00:09 --:--:--     0* Could not resolve host: www.tarantool.io
* Closing connection 0
curl: (6) Could not resolve host: www.tarantool.io
Error: Process completed with exit code 6.
```

--- a/.github/workflows/deploy-branch.yml
+++ b/.github/workflows/deploy-branch.yml
@@ -51,8 +51,15 @@ jobs:
 
       - run: bash upload_output.sh
         id: upload-output
-      - run: bash webhooks/update_deployment.sh
+
+      - name: Create or update a dev server deployment with version ${{env.DEPLOYMENT_NAME}}
+        uses: nick-fields/retry@v2
         id: update-deployment-webhook
+        with:
+          command: bash webhooks/update_deployment.sh
+          timeout_seconds: 15
+          retry_wait_seconds: 15
+          max_attempts: 3
 
       - name: update deployment status
         id: finalize-deployment

--- a/.github/workflows/deploy-custom.yml
+++ b/.github/workflows/deploy-custom.yml
@@ -91,8 +91,15 @@ jobs:
 
       - run: bash upload_output.sh
         id: upload-output
-      - run: bash webhooks/update_deployment.sh
+
+      - name: Create or update a dev server deployment with version ${{env.DEPLOYMENT_NAME}}
+        uses: nick-fields/retry@v2
         id: update-deployment-webhook
+        with:
+          command: bash webhooks/update_deployment.sh
+          timeout_seconds: 15
+          retry_wait_seconds: 15
+          max_attempts: 3
 
       - name: update deployment status
         id: finalize-deployment

--- a/.github/workflows/deploy-custom.yml
+++ b/.github/workflows/deploy-custom.yml
@@ -42,7 +42,7 @@ jobs:
       TARANTOOL_UPDATE_URL: ${{secrets.TARANTOOL_DEVELOP_UPDATE_URL}}
       VKTEAMS_BOT_TOKEN: ${{secrets.VKTEAMS_TARANTOOLBOT_TOKEN}}
     steps:
-      - uses: actions/checkout@v2
+      - uses: actions/checkout@v3
         id: checkout
         with:
           submodules: recursive

--- a/.github/workflows/destroy-deployment-manually.yml
+++ b/.github/workflows/destroy-deployment-manually.yml
@@ -26,9 +26,14 @@ jobs:
           echo "DEPLOYMENT_NAME=${{github.event.inputs.deployment_name}}" >> $GITHUB_ENV
           echo "workaround until actions/runner#655 is fixed"
 
-      - name: Remove dev server deployment at ${{env.DEPLOYMENT_NAME}}
+      - name: Remove dev server deployment with version ${{env.DEPLOYMENT_NAME}}
+        uses: nick-fields/retry@v2
         id: delete-deployment-webhook
-        run: bash webhooks/delete_deployment.sh
+        with:
+          command: bash webhooks/delete_deployment.sh
+          timeout_seconds: 15
+          retry_wait_seconds: 15
+          max_attempts: 3
 
       - name: Remove GitHub deployment at ${{env.DEPLOYMENT_NAME}}
         uses: strumwolf/delete-deployment-environment@v2

--- a/.github/workflows/destroy-deployment-manually.yml
+++ b/.github/workflows/destroy-deployment-manually.yml
@@ -15,7 +15,7 @@ jobs:
       TARANTOOL_UPDATE_KEY: ${{secrets.TARANTOOL_UPDATE_KEY}}
       VKTEAMS_BOT_TOKEN: ${{secrets.VKTEAMS_TARANTOOLBOT_TOKEN}}
     steps:
-      - uses: actions/checkout@v2
+      - uses: actions/checkout@v3
         id: checkout
         with:
           token: "${{ secrets.GITHUB_TOKEN }}"

--- a/.github/workflows/destroy-deployment.yml
+++ b/.github/workflows/destroy-deployment.yml
@@ -25,9 +25,14 @@ jobs:
         id: set-deployment-name
         run: echo "DEPLOYMENT_NAME=${GITHUB_HEAD_REF##*/}" >> $GITHUB_ENV
 
-      - name: Remove dev server deployment at branch-${{env.DEPLOYMENT_NAME}}
+      - name: Remove dev server deployment with version ${{env.DEPLOYMENT_NAME}}
+        uses: nick-fields/retry@v2
         id: delete-deployment-webhook
-        run: bash webhooks/delete_deployment.sh
+        with:
+          command: bash webhooks/delete_deployment.sh
+          timeout_seconds: 15
+          retry_wait_seconds: 15
+          max_attempts: 3
 
       - name: Remove GitHub deployment at branch-${{env.DEPLOYMENT_NAME}}
         uses: bobheadxi/deployments@v1

--- a/.github/workflows/destroy-deployment.yml
+++ b/.github/workflows/destroy-deployment.yml
@@ -16,7 +16,7 @@ jobs:
       TARANTOOL_UPDATE_KEY: ${{secrets.TARANTOOL_UPDATE_KEY}}
       VKTEAMS_BOT_TOKEN: ${{secrets.VKTEAMS_TARANTOOLBOT_TOKEN}}
     steps:
-      - uses: actions/checkout@v2
+      - uses: actions/checkout@v3
         id: checkout
         with:
           token: "${{ secrets.GITHUB_TOKEN }}"

--- a/.github/workflows/main.yml
+++ b/.github/workflows/main.yml
@@ -22,7 +22,7 @@ jobs:
       DEPLOYMENT_NAME: latest
       VKTEAMS_BOT_TOKEN: ${{secrets.VKTEAMS_TARANTOOLBOT_TOKEN}}
     steps:
-      - uses: actions/checkout@v2
+      - uses: actions/checkout@v3
         id: checkout
         with:
           submodules: recursive

--- a/.github/workflows/main.yml
+++ b/.github/workflows/main.yml
@@ -48,8 +48,15 @@ jobs:
 
       - run: bash upload_output.sh
         id: upload-output
-      - run: bash webhooks/update_deployment.sh
+
+      - name: Update production server deployment with version ${{env.DEPLOYMENT_NAME}}
+        uses: nick-fields/retry@v2
         id: update-deployment-webhook
+        with:
+          command: bash webhooks/update_deployment.sh
+          timeout_seconds: 15
+          retry_wait_seconds: 15
+          max_attempts: 3
 
       - name: Send VK Teams message on failures
         # bot token won't be accessible in the forks

